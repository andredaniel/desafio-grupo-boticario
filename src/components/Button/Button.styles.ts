import styled, { css } from 'styled-components'
import { theme } from '../../theme'
import { IButton } from './Button'

export const ButtonWrapper = styled.div<IButton>`
  padding: 0.5rem 0;
  width: 100%;

  a,
  button {
    background: ${theme.colors.secondary.LIGHT};
    border-radius: 5px;
    border: 1px solid ${theme.colors.secondary.MAIN};
    font-size: 1rem;
    padding: 0.7rem 1rem;
    position: relative;
    transition: background-color 200ms ease, border-color 200ms ease,
      color 200ms ease;
    width: 100%;

    &:focus {
      outline: none;
    }

    &:hover {
      background-color: ${theme.colors.secondary.CONTRAST};
    }
  }

  ${({ variant }) => {
    switch (variant) {
      case 'primary':
        return css`
          a,
          button {
            border: none;
            background: ${theme.colors.success.LIGHT};
            color: #fff;

            &:hover {
              background-color: ${theme.colors.success.MAIN};
            }
          }
        `

      default:
        break
    }
  }}
`
