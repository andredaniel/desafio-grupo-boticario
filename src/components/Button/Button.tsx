import React, { ButtonHTMLAttributes } from 'react'
import Ink from 'react-ink'
import { Link } from 'react-router-dom'
import { ButtonWrapper } from './Button.styles'

export type VariationTypes = 'primary' | 'secondary' | 'info'

export interface IButton extends ButtonHTMLAttributes<HTMLButtonElement> {
  label?: string
  variant?: VariationTypes
  href?: string
}

const Button: React.FC<IButton> = ({ children, ...props }): JSX.Element => {
  const { href, label, id, type, variant, ...rest } = props

  const component = (
    <ButtonWrapper variant={variant}>
      <button type={type || 'button'} id={id} {...rest}>
        {children || label}
        <Ink opacity={0.1} />
      </button>
    </ButtonWrapper>
  )

  return <>{href ? <Link to={href}>{component}</Link> : component}</>
}

export { Button }
