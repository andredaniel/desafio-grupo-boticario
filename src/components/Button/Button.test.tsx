import { render } from '@testing-library/react'
import React from 'react'
import { BrowserRouter as Router } from 'react-router-dom'
import { Button } from './Button'

describe('<Button />', () => {
  it('should render a primary button', () => {
    const { container } = render(<Button variant="primary" />)
    expect(container).toMatchSnapshot()
  })

  it('should render a secondary button', () => {
    const { container } = render(<Button variant="secondary" />)
    expect(container).toMatchSnapshot()
  })

  it('should render a info button', () => {
    const { container } = render(<Button variant="info" />)
    expect(container).toMatchSnapshot()
  })

  it('should render a button with href', () => {
    const { container } = render(
      <Router>
        <Button href="/" />
      </Router>
    )

    expect(container).toMatchSnapshot()
  })
})
