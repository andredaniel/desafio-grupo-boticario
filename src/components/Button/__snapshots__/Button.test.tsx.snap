// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render a button with href 1`] = `
<div>
  <a
    href="/"
  >
    <div
      class="sc-bdfBwQ fYPAfR"
    >
      <button
        type="button"
      >
        <canvas
          height="0"
          style="border-radius: inherit; height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
          width="0"
        />
      </button>
    </div>
  </a>
</div>
`;

exports[`<Button /> should render a info button 1`] = `
<div>
  <div
    class="sc-bdfBwQ ItcEY"
  >
    <button
      type="button"
    >
      <canvas
        height="0"
        style="border-radius: inherit; height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
        width="0"
      />
    </button>
  </div>
</div>
`;

exports[`<Button /> should render a primary button 1`] = `
<div>
  <div
    class="sc-bdfBwQ bkQapN"
  >
    <button
      type="button"
    >
      <canvas
        height="0"
        style="border-radius: inherit; height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
        width="0"
      />
    </button>
  </div>
</div>
`;

exports[`<Button /> should render a secondary button 1`] = `
<div>
  <div
    class="sc-bdfBwQ fYPAfR"
  >
    <button
      type="button"
    >
      <canvas
        height="0"
        style="border-radius: inherit; height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
        width="0"
      />
    </button>
  </div>
</div>
`;
